src/docs/guides/assets.md

```diff
@@ -0,0 +1,251 @@
+---
+title: Assets
+description: Learn how to reference assets in your components
+url: /docs/assets
+contributors:
+  - splitinfinities
+  - simonhaenisch
+  - rwaskiewicz
+---
+
+# Assets
+
+Stencil components may need one or more static files as a part of their design.
+These types of files are referred to as 'assets', and include images, fonts, etc.
+
+In this guide, we describe different strategies for resolving assets on the filesystem.
+
+> CSS files are handled differently than assets; for more on using CSS, please see the [styling documentation](/docs/styling).
+
+## Asset Base Path
+
+The **asset base path** is the directory that Stencil will use to resolve assets.
+When a component uses an asset, the asset's location is resolved relative to the asset base path.
+
+The asset base path is automatically set for the following output targets:
+- [dist](/docs/distribution)
+- [hydrate](/docs/hydrate-app)
+- [www](/docs/www)
+
+For all other output targets, assets must be [moved](/docs/assets#manually-moving-assets) and the asset base path must be [manually set](/docs/assets#setassetpath).
+
+For each instance of the Stencil runtime that is loaded, there is a single asset base path.
+Oftentimes, this means there is only one asset base path per application using Stencil.
+
+## Resolution Overview
+
+The process of resolving an asset involves asking Stencil to build a path to the asset on the filesystem. 
+
+When an asset's path is built, the resolution is always done in a project's compiled output, not the directory containing the original source code.
+
+The example below uses the output of the [`www` output target](/docs/www) to demonstrate how assets are resolved.
+Although the example use the output of `www` builds, the general principle of how an asset is found holds for all output targets.
+
+When using the `www` output target, a `build/` directory is automatically created and set as the asset base path.
+An example `build/` directory and the assets it contains can be found below.
+
+```
+www/
+├── build/
+│   ├── assets/
+│   │   ├── logo.png
+│   │   └── scenery/
+│   │       ├── beach.png
+│   │       └── sunset.png
+│   └── other-assets/
+│       └── font.tiff
+└── ...
+```
+
+To resolve the path to an asset, Stencil's [`getAssetPath()` API](/docs/assets#getassetpath) may be used.
+When using `getAssetPath`, the assets in the directory structure above are resolved relative to `build/`.
+
+The code sample below demonstrates the return value of `getAssetPath` for different `path` arguments.
+The return value is a path that Stencil has built to retrieve the asset on the filesystem.
+```ts
+// with an asset base path of "/build/":
+
+// '/build/assets/logo.png'
```

Comment:
Could we maybe put these comments inline with the code? It's a little confusing showing the result before the function call.

Reply:
We could, but unfortunately that leads to creating a code block that requires horizontal scrolling on a smaller device (even on my 13" laptop).  LMK what you think, but this felt like the more readable of the two

Reply:
Ah I didn't consider that. I prefer your version then.